src/gui/newwizard/SetupWizardController.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "pages/AbstractSetupWizardPage.h"
+#include <QDialog>
+#include <QSharedPointer>
+#include <SetupWizardAccountBuilder.h>
+#include <SetupWizardWindow.h>
+#include <account.h>
+
+namespace OCC::Wizard {
+/**
+     * This class is the backbone of the new setup wizard. It instantiates the required UI elements and fills them with the correct data. It also provides the public API for the settings UI.
+     *
+     * The new setup wizard uses dependency injection where applicable. The account object is created using the builder pattern.
+     */
+class SetupWizardController : public QObject
```

Comment:
Any gain of not inheriting SetupWizardWindow?
All I can see is the quite unusual showUi function?

Reply:
Separation of concerns. The controller does not do any UI-specific code, just business logic. The window and the pages implement UI-specific details. The builder serves as a data storage facility. Inheriting would violate SoC. I'd rather not mix both controller and frontend code.

The chosen architecture is similar to MVC.

Reply:
Then maybe make the widget accessible instead of implementing the showUi()